src/modules/flow/json/json.c

```diff
@@ -405,6 +405,21 @@ json_path_is_array_key(struct sol_str_slice slice)
     for (end_reason = SOL_JSON_LOOP_REASON_OK; \
         json_path_get_next_slice(&scanner, &key_slice, &end_reason);)
 
+static int32_t
+json_array_get_key_index(struct sol_str_slice key)
+{
+    long int index_val;
+    char *endptr;
+
+    errno = 0;
+    index_val = strtol(key.data + 1, &endptr, 10);
```

Comment:
@cabelitos did a safe version of this. Your version could overflow if key.data is not null-terminated. I'm not sure that was merged already.